### ISSUE TITLE
 Nano Adblocker / Defender for Chromium-based Edge issues mega thread

### ISSUE THREAD

**Issue by jspenguin2017** [NanoAdblocker/NanoCore#253]
Nano Adblocker is available here: https://www.microsoft.com/en-us/microsoft-edge/insider-addons/detail/epbkapkgcmdmfpogenoebpdeibmfinpf

Note: All extensions in the insider extensions store are tested and published by Microsoft, for any issues, you should report to Microsoft first.

**Comment by seba2282** [comment]
@jspenguin2017 I understand, you will add nano defender too, if it is chromium base. When you add? People still waiting.

**Comment by jspenguin2017** [comment]
The insider store is currently invite-only.

**Comment by seba2282** [comment]
@jspenguin2017, can you tell more details, how works that insider extension program? I saw Nano Adblocker.

**Comment by jspenguin2017** [comment]
@seba2282 What specifically?

**Comment by seba2282** [comment]
@jspenguin2017, you said about extension insider store. Can you more details, how extensions start be in that? Nano Adblocker is. What with Nano Defender?

**Comment by jspenguin2017** [comment]
@seba2282 I got an invitation email from Microsoft for Nano Adblocker, but didn't get one for Nano Defender.

**Comment by seba2282** [comment]
If you want informations, I know, Windows 10 19H1 will not have Edge on Chromium. Maybe will be in Windows 10 19H2. Important is, Windows 10 as separate will be on abonament or only with some services like xbox pass and office 365.

**Comment by lychichem** [comment]
The install package of Chromium-based Edge leaked yesterday. You can test with that package. As the program has a switch to "install extensions from other sources", I think upload Nano adblocker/defender will only be useful for users who cannot access Google services, for example, users in mainland China.

**Comment by jspenguin2017** [comment]
I'll wait for an official release to confirm.

**Comment by seba2282** [comment]
@jspenguin2017 and you can't give new versions Nano Adblocker to actual Microsoft Edge?

**Comment by jspenguin2017** [comment]
@seba2282 For Chromium-based Edge, Microsoft will be managing the extensions for now. For legacy Edge, see my comment here: https://github.com/NanoAdblocker/NanoCore/issues/40#issuecomment-473612973

**Comment by ivanmestre** [comment]
@jspenguin2017 it would be nice if support for older Edge could be maintained because, even though the new Edge will be out in a couple of weeks, it will be released as a preview which most regular users won't use. No one knows when the new Edge will be available as a stable build, it will probable be available either for the 1909 or 2003 build, so 6 months or 1 year where regular users won't receive any new Nano Adblocker versions.

**Comment by jspenguin2017** [comment]
@ivanmestre If enough people want another build for legacy Edge, I will reconsider. Cast your vote by adding 👍 to the comment above.

**Comment by ivanmestre** [comment]
Thanks, also Nano Adblocker works great on the new Edge. Both the one in the Edge Insider Extensions page and the one on the Chrome Store. I believe they are the same extension, but the one on the Edge Insider page is an older build. I believe what Microsoft did was listing the extensions that are currently in the Microsoft Store for old Edge and publishing the Chrome version on the Insider page. I think that's the case because Nano Adblocker on the Microsoft Store on old Edge is version 0.0.85.0 and the one in the Edge Insider page on new Edge is version 1.0.0.77.

**Comment by seba2282** [comment]
@jspenguin2017, Nano Adblocker will be continue for old MS Edge? Is few time before MS give final MS Edge Chromium-base.

**Comment by jspenguin2017** [comment]
@seba2282 I have explained it above: https://github.com/NanoAdblocker/NanoCore/issues/253#issuecomment-476933171

**Comment by seba2282** [comment]
Any information about few support for old Edge until time, when will be Final Edge Chromium?

**Comment by jspenguin2017** [comment]
@seba2282 There's only 4 upvotes on the post above, so chances are I won't bother with the old Edge anymore. The Chromium-based Edge should be ready in a few months, it's already out for public beta.

**Comment by seba2282** [comment]
@jspenguin2017, you didn't think about people who hasn't GitHub accounts? I guess is more people, who like your Nano products and they won't have actual any insiders Edge or other insider products and they wait for final version as regular. You only look on people who active on GH. But on others who haven't accounts GH...

**Comment by jspenguin2017** [comment]
@seba2282 Obviously I'm not expecting hundreds of upvotes. Even though most people don't have a GitHub account, 4 votes is still too little. I know the new Edge is not yet bundled with Windows, but people can just go download it, I don't see how is that a problem.

**Comment by PeteAtGH** [comment]
Please keep the Nano Adblocker for EgdeHTML up to date. The new Chromium based one is awful on any touch & pen devices. It may/will take a very long time to match the usability of the old Edge, if ever at all. Who knows what's going to happen with the new incarnation of Edge, where the old one is perfectly fine and way better. Thank you.

**Comment by jspenguin2017** [comment]
Some work is needed on Edgyfy side. I need an up-to-spec polyfill for `TextEncoder` / `TextDecoder`.

**Comment by gdata1** [comment]
Hello. When will the 1.00.104 version be available in the Microsoft store?

**Comment by seba2282** [comment]
When @jspenguin2017 give update. So after 6-12 months. Because Edge Chromium will after that time as stable and integrate with Windows 10. Author still isn't interesting old Edge. 
Sorry for true.

**Comment by gdata1** [comment]
I'm talking about an update for the new edge because I use it since the day of the release for insiders.

**Comment by PeteAtGH** [comment]
Dear **jspenguin2017**, thank you for taking into consideration the option of updating the Nano Adblocker for EdgeHTML once there are 20 votes for it. Hopefully happens soon. I think  Microsoft will never give up completely on EdgeHTML, at least not in the near future. I'm all Surface, same many of my friends and a proper touch & pen  support is indispensable for us.

**Comment by jspenguin2017** [comment]
@gdata1 The extension store for the new Edge is fully managed by Microsoft, and I can't update the listing. Note that you can install extensions directly from the Chrome WebStore to get the latest version.

**Comment by gdata1** [comment]
jspenguin2017 Do I understand that you can not tell Microsoft about the release of the new version?

**Comment by jspenguin2017** [comment]
@gdata1 I don't see a way beside emailing them. I highly doubt they read the emails though, you can try: `edgeext@microsoft.com`

**Comment by gdata1** [comment]
@jspenguin2017 I sent them a letter. Will wait for a response from them

**Comment by gdata1** [comment]
@jspenguin2017 For the new Edge came the update 1.00.104

**Comment by seba2282** [comment]
@jspenguin2017, support for Edge with old source dead permanently with updates?

**Comment by jspenguin2017** [comment]
@seba2282 I have answered it here: https://github.com/NanoAdblocker/NanoCore/issues/253#issuecomment-483473077

**Comment by PeteAtGH** [comment]
Dear **jspenguin2017**,  I cannot believe there are only 9 upvotes for an EdgeHTML update. The new Chromium Edge is a pile of crap on any Surface and it will take Microsoft years to make it work, or never at all. Hopefully they will realise it sooner than later and abandon it in favour or EdgeHTML. Please reconsider again your decision regarding EdgeHTML. Thank you very much for all your hard work you put in so far.

**Comment by seba2282** [comment]
Well is better alternative for Ublock Origin and his fork. Named Adguard. Team Adguard care about users and update their Adblocker for MS Edge on old source. And they can. More devs Adblockers gone for easy way (chromium). Sorry for true @jspenguin2017.

**Comment by jorgegonzalez** [comment]
@PeteAtGH Have you even tried the new Chromium Edge? It runs like butter.

**Comment by PeteAtGH** [comment]
**@seba2282** Thank you for your advice but I know Adguard AdBlocker and used it for a while after uBlock Origin for EdgeHTML stopped updating. It's quite good, 2nd on my list behind Nano Adblocker. Let's hope Nano gets maintained after all at the mercy of the creator?

**Comment by PeteAtGH** [comment]
**@jorgegonzalez** Yes I have, it's awful, slow, resource hungry, almost non-existed touch and pen support, bad scrolling, crashing, drains battery in connected standby when left open, extensions don't update in the background like on EdgeHTML when closed, etc. Tested on various MS Surfaces all with LTE.

**Comment by jorgegonzalez** [comment]
@PeteAtGH If you say so, but that has not been my experience.

**Comment by jspenguin2017** [comment]
Hum... Looks like gorhill added a fallback for text encoder, I should be able to roll out an update if everything goes as planned. 
https://github.com/gorhill/uBlock/commit/295f08da9790da7ba9f1dc79f443e9df5756e42f

**Comment by jspenguin2017** [comment]
I'll need to wait until 1.18.18 though, so it's probably not happening within a few weeks.

**Comment by PeteAtGH** [comment]
Dear **@jspenguin2017**, it's much appreciated, thank you very much. Will wait patiently, better late than never. 👍

**Comment by seba2282** [comment]
@jspenguin2017 👍

**Comment by seba2282** [comment]
@jspenguin2017, is Ublock 1.19 
Any information about support on old Edge on  EdgeHtml ? Will you add some update your extension?

**Comment by PeteAtGH** [comment]
Dear **@jspenguin2017**, as you're aware uBlock Origin 1.19.0 is out. How likely is an updated Nano release for EDGEHtml and if, what's the time frame, please? Have a nice weekend.

**Comment by jspenguin2017** [comment]
@seba2282 @PeteAtGH Yes, I saw that yesterday. I will test the new build to see how well it works on the legacy Edge, I'll publish a new build if it looks good. This is a personal project so I will work on it when I get to it.

**Comment by seba2282** [comment]
@jspenguin2017 thank you. And you need know something. Old version doesn't block cookies, downloaders and more on dobreprogramy.pl in EdgeHtml. Will be fix with new Nano Adblocker?

**Comment by jspenguin2017** [comment]
@seba2282 Probably not, that sounds like a filters issue.

**Comment by jspenguin2017** [comment]
Hum... It looks like the new build is working alright on the legacy Edge. The Logger is not working but I don't see any other issues. I'll publish a new build.

**Comment by PeteAtGH** [comment]
**@jspenguin2017** much appreciated. The logger neither works on the current version. any chance to have a go at it in the near future, please? Thank you.

**Comment by jspenguin2017** [comment]
@PeteAtGH If I can find a good polyfill for legacy Edge's broken `querySelector` then maybe... You can just use F12 DevTools, I don't consider it a big issue.

**Comment by seba2282** [comment]
@jspenguin2017, you didn't think over publicate nano Defender for Edge legacy? Mean in Microsoft Store? 🙂

**Comment by PeteAtGH** [comment]
**@jspenguin2017** no worries, thank you for the tip. Many thanks again for all your efforts.

**Comment by PeteAtGH** [comment]
**@seba2282** an updated Nano Adblocker has been published nearly 1 hour ago in the Microsoft Store.

**Comment by seba2282** [comment]
@PeteAtGH, installed, Actual in my country is older version.

@jspenguin2017, you wondered about public Nano Defender?

**Comment by jspenguin2017** [comment]
@seba2282 The answer to your question is in [the docs](https://github.com/NanoAdblocker/NanoCore/blob/master/notes/install-in-edge.MD#store-version).

**Comment by seba2282** [comment]
@jspenguin2017, I understand all. But @PeteAtGH wrote about is update in Microsoft Store for nano adblocker. So qustion, when nano defender will be too in MS store?

**Comment by jspenguin2017** [comment]
@seba2282 I have answered your question here: https://github.com/NanoAdblocker/NanoCore/issues/253#issuecomment-491541795

**Comment by jspenguin2017** [comment]
I submitted an update, not sure if Microsoft is still accepting them though.

**Comment by gdata1** [comment]
@jspenguin2017 Can send an update for the new edge?

**Comment by jspenguin2017** [comment]
@gdata1 No, I don't have access to the listing. You can just get it from Chrome Web Store though.

**Comment by PeteAtGH** [comment]
**@jspenguin2017** In the Microsoft Store the time stamp of the most recent update is showing 12/05/2019 02:43:46 at present.

**Comment by seba2282** [comment]
@PeteAtGH, I don't see that still in MS Store, what you said.

**Comment by jspenguin2017** [comment]
@seba2282 I got an email asking me to update the screenshots, which are definitely too out of date by now. It shouldn't take too long, hopefully I can get it done today or tomorrow.

**Comment by jspenguin2017** [comment]
I just re-submitted the update.

**Comment by PeteAtGH** [comment]
**@jspenguin2017** - MANY BIG THANKS. The long expected updated Nano Adblocker
 0.0.109.0 is available to install from MS Store. You are a Gentleman.
**@seba2282** - Enjoy !!!

**Comment by seba2282** [comment]
@jspenguin2017, thank you. 👍
So now waiting on Nano Defender. Edge on Chromium will be probably near ending year or start 2020. So please look on my suggest positive and give ND finally to MS Store. 
Thank you. 🙂

**Comment by jspenguin2017** [comment]
@seba2282 I already submitted Nano Defender to Microsoft Store last year, but it's still in queue. That's out of my control.

**Comment by seba2282** [comment]
@jspenguin2017, ahh... So you don't know why Microsoft won't accept your defuser for adblocker and put to Store. If I good understood.

**Comment by jspenguin2017** [comment]
I see the problem, legacy Edge does not support `:scope`.

**Comment by jspenguin2017** [comment]
Looks like this can be easily fixed by a clever hack by `jonathantneal`. I'll ship a fix.

**Comment by seba2282** [comment]
Nano still has problem with hidden Avast promo and dowbloader on dobreprogramy. But Adguard Team fixed problem with Rodo/cookies on filters. Maybe time fix that not blocking Avast promo and dowbloader after refresh site? Nano should hide that elements normal with filters. But something in code make doesn't hide that.

**Comment by jspenguin2017** [comment]
@seba2282 OK, I see the problem. I'm not sure why it's happening but neither the official Edge port nor Nano can hide those ads. It does work on Chrome though, which is kind of weird.

**Comment by PeteAtGH** [comment]
**jspenguin2017** You're a HERO. Today's update fixed "The logger". Amazing. Thanks a lot.

**Comment by seba2282** [comment]
@jspenguin2017, you fixed problems with cookies and hiding Avast promo with downloader on dobreprogramy.pl ? What I reported you for Edge on EdgeHtml?

**Comment by jspenguin2017** [comment]
@seba2282 I'm not sure what's causing the issue. I see some errors thrown from browser internal code. I still need to investigate.

**Comment by seba2282** [comment]
@jspenguin2017, but you found what making that errors in browsers? Adguard needed one rule fix and only that Adblocker actually hasn't problem on MS Edge EdgeHtml. I don't know, how they make that, but seriously it work. About nano/ublock origin, I have problem with them on old Edge and I don't know, why. They won't block me on Edge cookies/rodo, ads and others. Is something bad. @gorhill hasn't any fix?

**Comment by jspenguin2017** [comment]
@seba2282 Nope, no clue what is causing the error, the stack trace is empty. Looks like it's coming from native code.

**Comment by seba2282** [comment]
@jspenguin2017, you should contact with someone from Adguard Team too for ask how fix problems. You all are developers. Adguard Team always like help and they are really good. Maybe try contact with them. 🙂

**Comment by jspenguin2017** [comment]
@seba2282 No, that doesn't make sense. The architecture of the two extensions are vastly different.

**Comment by seba2282** [comment]
@jspenguin2017, maybe go merge Nano Adblocker with Nano Defender in one extension and bundle scripts and rules filters? Adguard will back script filters to Adguard 3.1 and again will be great skip between anti adblocks and all again will works great. Maybe try that way. 🙂

**Comment by PeteAtGH** [comment]
**@seba2282** if I may say something, please don't get me wrong but none of the various adblockers I've ever used is perfect with every browser. Nano has been the best one for me so far in many different environments like work & private and Adguard is breaking far more sites than Nano, requires more resources and slows down browsers. After all,  it's still a viable option in case Nano retires. There is another browser I use every now and then called Monument Browser, available from MS Store. It has got a build-in ad blocker and anti social and tracking filter, which you can enable in Extra Features. I tried the website you mentioned before "http://www.dobreprogramy.pl" and didn't see a single ad.

**Comment by jspenguin2017** [comment]
@seba2282 Nano Defender cannot be made configurable with current APIs, so that can't be done.

**Comment by seba2282** [comment]
Problem is that, if look on Nano with UBO and Adguard. Adgurd blocks actually better on old Edge. After add rule for blocking cookies/rodo on domains wp.pl, ada still aren't blocking on dobreprogramy.pl example. If use Adguard (I reported that bug with cookies and they added very good rule for fix), he blocks all. More people say, Adguard team use better scripts and rules in Adguard and that works very good. To version 3.1 will come back scripts rules filters so will better like earlier. So @jspenguin2017, I gave you great suggest earlier, what is possible. 
@PeteAtGH, that what Adguard break, it is by filters what you turned. There are filters what tagged as obsolete/problematic. You don't need turn them. About memory or browser slower, I don't know why is that. On Android Firefox I know AG use much memory, so actually use app Android. On PC I don't see to much slow browser. But... Team AG promised future optimization memory.

**Comment by kevin12314** [comment]
Add custom filter
`dobreprogramy.pl##+js(nano-remove-elements-onready.js, #phContent_avastBadge)`
and update Windows to 19H1, Avast promo gone.
![image](https://user-images.githubusercontent.com/10508800/58239007-d7b5aa80-7d7a-11e9-8cbc-2953cfcac589.png)

**Comment by seba2282** [comment]
Works after refresh site? If you visited dobreprogramy.pl, you had cookies to accept and downloader some program if you not logged? After refresh site, I know, didn't work nothing. But if 19H1 fix problem, then try that after moment. Thank you ☺️

**Comment by jspenguin2017** [comment]
@seba2282 If you think you have a great suggestion, feel free to test it out and send a PR. We can discuss and test it further.

**Comment by PeteAtGH** [comment]
**@jspenguin2017** Many thanks for keeping Nano updated. Everything works fine apart from when attempting to open Nano settings in InPrivate mode  "HTTP 501 error  This website can’t or won’t display the page you want" - any idea? Tried on several PCs with same outcome. Thanks.

**Comment by jspenguin2017** [comment]
@PeteAtGH Side effect of spanning mode. Split mode has bigger issues. One workaround is to open a normal window instead of relying on `runtime.openOptionsPage`. I'm quite busy recently so a PR for Edgyfy is welcomed.

**Comment by seba2282** [comment]
I suggest read that article (few without Edge Chromium topic) but about manifest api 3.0 for chromium: 
https://www.theregister.co.uk/2019/05/29/google_webrequest_api/

**Comment by jspenguin2017** [comment]
@seba2282 For Manifest V3 related discussion, go here: https://github.com/NanoAdblocker/NanoCore/issues/238

**Comment by jspenguin2017** [comment]
There's a bug in version 116 that can be triggered by Adguard filters. Version 117 is in certification queue. 
If you are on default settings and your browser language is not Russian, you're not affected.

**Comment by PeteAtGH** [comment]
**@jspenguin2017** The filter lists stopped updating or being modified in version 0.0.123.0 for EdgeHtml on all devices I supervise. Please help.

**Comment by jspenguin2017** [comment]
Looks like the CDN hosting the filters is having some issues, I'll investigate.

**Comment by jspenguin2017** [comment]
I reproduced the issue with filters update, and it's not caused by the CDN server issues. I'm not sure why this is happening and I don't have time to debug this. Contribution welcomed. 

I highly recommend everyone still using legacy Edge to upgrade to the new Edge as it's clear that I won't be able to provide support for legacy Edge for long.

**Comment by PeteAtGH** [comment]
**@jspenguin2017** PLEASE , PLEASE try to fix the lists' updating issue for EdgeHTML. There is constantly something wrong with the new Edge, it's not usable, please have a look:  https://techcommunity.microsoft.com/t5/Discussions/Updated-Dev-channel-update-to-78-0-268-1-is-live/m-p/847300 , many sites are not being rendered properly, keeps crashing daily.
The new Edge is just crap, same as Chrome/Chromium. Microsoft's worst decision ever.

**Comment by jspenguin2017** [comment]
I don't have time to debug this right now. If the previous version works, I'll do a rollback.

**Comment by jspenguin2017** [comment]
I can't reproduce this issue anymore in the latest release. I'm publishing an update.

**Comment by jspenguin2017** [comment]
The new release is still in certification.

**Comment by PeteAtGH** [comment]
**@jspenguin2017** you're a MAGICIAN, all work a treat. Thank You very much.

**Comment by jspenguin2017** [comment]
What do you mean?

**Comment by PeteAtGH** [comment]
@jspenguin2017 It's all fine now, the update became available sometime early afternoon today and woks smoothly. Thank you ever so much.

**Comment by xlollomanx** [comment]
@jspenguin2017 Some days ago I reported an adblock detector from this test link: 

> https://www.vvvvid.it/show/892/sword-art-online-alicization-war-of-underworld/1005/541010/nelle-terre-del-nord

here: https://forums.lanik.us/viewtopic.php?f=96&t=29982&sid=c9e5be4508ebe1964b3cca59844216f2&start=60

They fixed it in no time. After a while I noticed that the fix works beautifully on Chrome, but when I try it on Edge chromium it works weird: it took long time to load the video and sometimes nanoadblocker/defender is still detected. I don't think is site related because i tried to open that link both on chrome and firefox with same filters and they works fine.

**Comment by jspenguin2017** [comment]
@xlollomanx This might be sticky, see: https://github.com/NanoMeow/QuickReports/issues/149

**Comment by xlollomanx** [comment]
@jspenguin2017 Maybe I was a bit confusionary. At the moment vvvid site is working fine for me on chrome, any ads was skipped and anti-adblock was killed since it has been fixed in ublock list. When I try to open the same page with chromium Edge (same filters and adblock settings) it takes long time to load the page and sometimes the page "please disable your adblock" triggers. This sound very weird for me because on chrome works very good and edge should be very similar. Seems like a sort of regression on edge. I forgot to mention I'm currently using the last leaked stable channel build 79.0.309.18. 
After some test I got vvvid working everytime on edge (not necessary on chrome) after I added this custom filter `@@||vast.doubleverify.com/v3/vast$xhr,domain=imasdk.googleapis.com`

**Comment by jspenguin2017** [comment]
Hum... It could be a bug in Edge. Can you try the other builds (beta, dev)?

**Comment by xlollomanx** [comment]
I tried latest canary build 80.0.331.0, I have the same result: sometimes "disable adblock please" triggers and when it works it toake long time compared to chrome. I tested even with ublock+nano defender and it's the same. Now I don't know what and to who I have to report it and what I have to write in the report...

**Comment by jspenguin2017** [comment]
What about beta and dev builds?

**Comment by xlollomanx** [comment]
I've just loaded the test link `https://www.vvvvid.it/show/892/sword-art-online-alicization-war-of-underworld/1005/541195/la-vigilia-della-guerra` both on dev and beta build. Both of them takes long time to load the video and both show randomly the message "disable adblock please".

**Comment by jspenguin2017** [comment]
Hum... That's weird, I'll look into this when I have time.

**Comment by PeteAtGH** [comment]
Dear **jspenguin2017** , the current EdgeHTML MS Store version 0.0.128.0 works beautifully amazing !!! But is there any particular reason for version 0.0.130.0 not being published in the MS Store?
By the way, when trying both links "https://www.vvvvid.it/show/892/sword-art-online-alicization-war-of-underworld/1005/541195/la-vigilia-della-guerra" and "https://www.vvvvid.it/show/892/sword-art-online-alicization-war-of-underworld/1005/541010/nelle-terre-del-nord", the following happens:
- on EdgeHTML none of them loads fully due to excessive amount of items (45 -almost all of them linked to Google owned domains) being blocked by Nano and I'm VERY happy with it, no need for Google's tracking,
- on Firefox Nano only blocks 1! item, the pages load with a warning to disable adblock, not happy with it as Nano appears BROKEN allowing all the other items through, which are blocked on EdgeHTML.
On both browsers Nano is configured in identical way.
Please keep the good job on EdgeHTML, a browser way better than any incarnation of Chromium.
Beside, there was a major problem with Google Chrome last week, " https://www.bleepingcomputer.com/news/software/google-fixes-white-screen-problem-in-chrome-admins-furious/ " which affected Edge Beta as well, fortunately only in our testing environment, so no one got impacted by it.

**Comment by xlollomanx** [comment]
@jspenguin2017 my issues described above has been fixed on recent filters update. Now for me it works in the same way on firefox, chrome and chrome edge.

**Comment by jspenguin2017** [comment]
@PeteAtGH 
I didn't have time to test the build on Edge, there were a lot of changes and testing was taking a lot of time. If you're willing to help with testing, the extension package can be downloaded from the release page. 

There were recent commits in uAssets regarding VVVVID (3h ago from time of this post), try updating your filters, if you're still experiencing issues, submit a report with the Quick Issue Reporter or open a new issue here. 

Also keep in mind that I don't maintain the Firefox build, for Firefox specific issues, report here: https://github.com/LiCybora/NanoCoreFirefox/issues

The recent issue with WebContent Occlusion only affect multi-user server environments, which didn't affect me and I'm not bothered by it. By contrast, EdgeHTML debugger is really buggy and that heavily affects me when I'm testing extensions.

**Comment by PeteAtGH** [comment]
**@jspenguin2017** Apologies for my late reply caused by a very busy period as usual before Christmas.
Tried the new version 0.0.130.0 on EdgeHTML, unfortunately it's plagued with major issuest, it does load but without any options enabled and missing all filters. Manually ticked options are not saved, lists cannot be added and restore button doesn't trigger any action. Effectively it's as good as dead.
Hopefully you'll be able to figure out what's wrong, please.
Back to the issue I mentioned about Chrome/Edge C, in our case not only RD users were affected, also few single user PCs/machines got hit as well with a white screen of dead Chrome. Reverting to an older version fixed the trouble and luckily all of them are test environments.
Last week I spent some time testing the new EdgeDev, the thing is broken beyond any usable scenarios, reported all of them to Edge insider. Major problems were: breaks Windows Defender Application Guard, freezes on Surface Type Cover close, doesn't suspend properly - drains battery, loses internet connection on all Surface LTEs while travelling, breaks face recognition on Surface tablets, unable to correctly read Surface GPS sensors reporting wrong locations and doesn't update it while moving/travelling  around, freezes terribly using Google maps, works better with Bing maps, etc. I doubt the new Edge C is going to be ready for the announced stable release on 15/01/2020, if ever.
Therefore please keep maintaining your extension for EdgeHTML if possible.

**Comment by jspenguin2017** [comment]
@PeteAtGH 
I have final exams coming up and I probably will have a lot of stuff to do over winter break too. So unless someone wants to investigate the issue and submit a patch, this will have to wait for a while. 

If you're using my projects in an enterprise environment, then you really need to consider hiring a freelancer to contribute back instead of relying on my volunteer work.

**Comment by kevin12314** [comment]
Microsoft Edge Addons store is now open for submissions today.
[Get started building extensions for the new Microsoft Edge](https://blogs.windows.com/msedgedev/2019/12/16/get-started-extensions-addons-microsoft-edge-chromium/#fFKvWPuwKJ6ewlEy.97)

[Publish An Extension](https://docs.microsoft.com/en-us/microsoft-edge/extensions-chromium/publish/publish-extension?branch=extensions-documentation)

Hope to see Nano Defender in new Microsoft Edge Addons store.

**Comment by jspenguin2017** [comment]
@kevin12314 
It opened for submission on the 5th, and Nano Defender is already published. I make it unlisted as I still have to do some testing. 

If you want to get it now: https://microsoftedge.microsoft.com/addons/detail/ijfkmnlofajajikjhfiigelipempcklj

**Comment by kevin12314** [comment]
@jspenguin2017 
Excellent! I'll install it and try it.

**Comment by TitusTroy** [comment]
ever since updating to Windows 10 version 1909, Nano Adblocker no longer seems to work (the old one not the new Chromium based one)...is it just me or is it a known issue?

**Comment by jspenguin2017** [comment]
@TitusTroy The version for legacy Edge is no longer maintained. You should upgrade to the new Chromium based Edge.

**Comment by TitusTroy** [comment]
> @TitusTroy The version for legacy Edge is no longer maintained. You should upgrade to the new Chromium based Edge.

thanks for the info...so the older version doesn't work at all anymore?...I was going to wait a few weeks before using Chromium Edge in case there were any bugs...so a new version of Nano Adblocker is available for Chromium Edge?

**Comment by kevin12314** [comment]
I tested Nano Adblocker  on the Legacy Edge and it looks like it's working. 
Edge version：Microsoft Edge 44.19546.1000.0
Microsoft EdgeHTML 18.19546

**Comment by jspenguin2017** [comment]
@TitusTroy 

I don't know if the older version still works or not, it should still work unless something changed in legacy Edge. 

There is a newer version of NA (and ND) available for Chromium based Edge, I have tested it and it works properly.

**Comment by TitusTroy** [comment]
so after I download the new Chromium Edge do I still need to go to the Microsoft Store to find the new version of Nano Adblocker?

**Comment by jspenguin2017** [comment]
@TitusTroy It should ask you, if not, you may have to install it manually. Note that this is not a tech support forum, for general questions about how to use Edge, you should search on Google first.

**Comment by PeteAtGH** [comment]
@jspenguin2017 That's fine, I understand. The current version for EdgeHTML works perfectly fine so I'll be patiently waiting for an updated version.
I don't use your project in an enterprise, it's on much smaller scale in few SBes, which couldn't afford doing as you've suggested. The new Edge doesn't cut for them, they just hate it, so do I. They're Surface based in highly touch, pen and mobile environment. Please keep your project alive as long as you can for EdgeHTML, which will be supported by Microsoft still for several years to come.

**Comment by jspenguin2017** [comment]
@PeteAtGH 

Even though I am still in university, my time is not cheap. Instead of thanking me for creating these projects, you decided to complain about terminating support of a dead platform. That is not nice. You cannot afford to hire a freelancer to keep a project you use maintained is not a reason for me to spend my time maintaining it for free. 

I repeat what I said, unless someone step up to contribute to the maintenance of the legacy Edge build, it will no longer receive updates.

**Comment by PeteAtGH** [comment]
@jspenguin2017 

Your time, your business, your problem, your decision, your project. I never complained, you must have read someone else's comments, that's not fair, I always expressed my gratitude for your work, check my posts. And the "platform" is not dead, at least not yet, still a default browser on every current and insider build, incl. W10 X emulator, W10 S, HoloLens 2, Xbox, etc. Works properly with Office 365 online, SharePoint, variety of corporate and specialist websites, etc., where the new ChrappyEdge fails. Have a look on Microsoft's domestic and enterprise forums at the amount of negative posts about the NewEdge. Nevertheless, your choice but many thanks again for creating this project and all your hard work as well as time you put into it. Cheers.

**Comment by jspenguin2017** [comment]
@PeteAtGH 

It is my time, my decision, and my project. However, it is your business and your problem. I do not use the legacy Edge anymore and the new Edge works fine for me. I ran into far more issues with the legacy Edge and I am happy to ditch it as soon as the new Edge came out. 

In your previous comments, you are pushing me to release an update for legacy Edge. I took it as a complain but it does not matter, it is a complain or not will not change the current situation. 

My decision is final, unless another maintainer step up to contribute, the legacy Edge build will not be updated.

**Comment by seba2282** [comment]
@jspenguin2017 Tell true, is easier make updates for chromium. What you will make when Manifest 3 finally welcome for Chromium and will be obligation in all browsers basic Chromium? You wondered about that scenario?

**Comment by jspenguin2017** [comment]
@seba2282 

Yes, it is way easier to develop for Chromium, the debugger just works and never crashes, unlike the legacy Edge or Firefox. 

As for dealing with Manifest V3, there are many potential solutions proposed in the [Manifest V3 megathread](https://github.com/NanoAdblocker/NanoCore/issues/238). Also, Chromium forks are not forced to take all changes from the upstream, so it is very possible that some forks will keep the old `webRequest` API. In fact, the developers of multiple Chromium forks have publicly announced that they intend to keep support for the old `webRequest` API. 

I certainly considered the scenario where all Chromium browsers moved to strictly Manifest V3, some solutions discussed in the megathread account for this worst case scenario.

**Comment by seba2282** [comment]
@jspenguin2017 Well but I don't trust to end Google corporation. They can make chaos with all Chromium project if they obligation will want for all Manifest 3. But...

**Comment by jspenguin2017** [comment]
@seba2282 The Chromium project is open source, so Google is not able to force Chromium forks to follow their rules. Let's stay on topic, for further discussion about Manifest V3, please use the other megathread.

**Comment by seba2282** [comment]
@jspenguin2017 about Nano Adblocker, you can give mechanism which can remove all frustrated eu cookies from sites?

**Comment by llacb47** [comment]
There are already anti-cookie notice lists, if that's what you are talking about

**Comment by seba2282** [comment]
@llacb47 www.dobreprogramy.pl without extension I don't Care About Cookies, Ublock Origin/Nano Adblocker don't remove cookies. Suggest try that site.

**Comment by llacb47** [comment]
Try `||wp.pl^$script,domain=www.dobreprogramy.pl`, but this is off-topic.

**Comment by jspenguin2017** [comment]
@seba2282 Nano has some abilities to remove cookie notices, but that is not the focus. You can open new issues in the [Nano Filters repository](https://github.com/NanoAdblocker/NanoFilters/issues) regarding cookie notices but those will be low priority. This is getting off topic, open new issues for other general questions.

**Comment by seba2282** [comment]
@llacb47 I try when I be on PC

@jspenguin2017 with fully respect. But your product after be few times and using by users, did can have special mechanism on remove cookies. Maybe to dedicated nano filters you give? Be in true. Cookies in EU are seriously annoying. I don't know how look that in USA and others continents. And I write in topic about Nano products. That function look nice for European user. Please look on my suggest. ☺

**Comment by dhimanAdhikari** [comment]
I cannot access/open any websites when Nano AdBlocker is turned on in Microsoft Edge Chromium.
When I turn off the extension I can open sites again. This started happening yesterday; was working just fine before.

I also have Nano Defender Pro intalled.

**Comment by jspenguin2017** [comment]
@dhimanAdhikari That's likely due to data corruption, a reinstall should fix the issue.

**Comment by jspenguin2017** [comment]
@dhimanAdhikari That can happen when the installation files are corrupted. A reinstall usually corrects the issue.
